static/app/components/modals/inviteMembersModal/inviteRowControlNew.tsx

```diff
@@ -0,0 +1,227 @@
+import {useCallback, useState} from 'react';
+import type {MultiValueProps} from 'react-select';
+import type {Theme} from '@emotion/react';
+import {useTheme} from '@emotion/react';
+import styled from '@emotion/styled';
+
+import type {StylesConfig} from 'sentry/components/forms/controls/selectControl';
+import SelectControl from 'sentry/components/forms/controls/selectControl';
+import {useInviteMembersContext} from 'sentry/components/modals/inviteMembersModal/inviteMembersContext';
+import RoleSelectControl from 'sentry/components/roleSelectControl';
+import TeamSelector from 'sentry/components/teamSelector';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import type {SelectValue} from 'sentry/types/core';
+import type {OrgRole} from 'sentry/types/organization';
+
+import renderEmailValue from './renderEmailValue';
+import type {InviteStatus} from './types';
+
+type SelectOption = SelectValue<string>;
+
+type Props = {
+  roleDisabledUnallowed: boolean;
+  roleOptions: OrgRole[];
+};
+
+function ValueComponent(
+  props: MultiValueProps<SelectOption>,
+  inviteStatus: InviteStatus
+) {
+  return renderEmailValue(inviteStatus[props.data.value], props);
+}
+
+function mapToOptions(values: string[]): SelectOption[] {
+  return values.map(value => ({value, label: value}));
+}
+
+function InviteRowControl({roleDisabledUnallowed, roleOptions}: Props) {
+  const {inviteStatus, pendingInvites, setEmails, setRole, setTeams, reset} =
+    useInviteMembersContext();
+  const emails = [...(pendingInvites.emails ?? [])];
+  const role = pendingInvites.role ?? '';
+  const teams = [...(pendingInvites.teams ?? [])];
+
+  const onChangeEmails = (opts: SelectOption[]) => {
+    setEmails(opts?.map(v => v.value) ?? [], 0);
+  };
+  const onChangeRole = (value: SelectOption) => setRole(value?.value, 0);
+  const onChangeTeams = (opts: SelectOption[]) =>
+    setTeams(opts ? opts.map(v => v.value) : [], 0);
+
+  const [inputValue, setInputValue] = useState('');
+
+  const theme = useTheme();
+
+  const isTeamRolesAllowedForRole = useCallback<(roleId: string) => boolean>(
+    roleId => {
+      const roleOptionsMap = roleOptions.reduce(
+        (rolesMap, roleOption) => ({...rolesMap, [roleOption.id]: roleOption}),
+        {}
+      );
+      return roleOptionsMap[roleId]?.isTeamRolesAllowed ?? true;
+    },
+    [roleOptions]
+  );
+  const isTeamRolesAllowed = isTeamRolesAllowedForRole(role);
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLElement>) => {
+    switch (event.key) {
+      case 'Enter':
+      case ',':
+      case ' ':
+        handleInput(inputValue);
+        setInputValue('');
+        event.preventDefault();
```

Comment:
nit: recommend `event` being `e` or `evt` (window.event is a reserved keyword in js, it's the latest event triggered) - ES6 scoping should correctly encapsulate the variable so it should be okay.

Oh, and it's generally recommended to have `preventDefault` as the first thing in a handler if you want to prevent it. (the idea being if there's an error in `setInputValue`, then the default handler would still be triggered)